e2e/test/iothub/NoRetryE2ETests.cs

```diff
@@ -55,81 +59,100 @@ await FaultInjection
 
             await Task.Delay(FaultInjection.DefaultDelayInSec).ConfigureAwait(false);
 
-            Logger.Trace($"{nameof(FaultInjection_NoRecovery)}: waiting fault injection occurs...");
-            for (int i = 0; i < FaultInjection.LatencyTimeBufferInSec; i++)
+            Logger.Trace($"{nameof(FaultInjection_NoRetry_NoRecovery_OpenAsync)}: waiting fault injection occurs...");
+            var sw = new Stopwatch();
+            sw.Start();
+            while (sw.Elapsed < FaultInjection.LatencyTimeBufferInSec)
             {
                 if (connectionStatusChanges.ContainsKey(ConnectionStatus.Disconnected))
                 {
                     break;
                 }
                 await Task.Delay(TimeSpan.FromSeconds(1)).ConfigureAwait(false);
             }
+            sw.Reset();
 
-            Assert.AreEqual(ConnectionStatus.Disconnected, lastConnectionStatus, $"Excepeted device to be {ConnectionStatus.Disconnected} but was {lastConnectionStatus}.");
-            Assert.IsFalse(connectionStatusChanges.ContainsKey(ConnectionStatus.Disconnected_Retrying), $"Shouldn't get {ConnectionStatus.Disconnected_Retrying} status change.");
+            lastConnectionStatus.Should().Be(ConnectionStatus.Disconnected, $"Expected device to be {ConnectionStatus.Disconnected} but was {lastConnectionStatus}.");
+            lastConnectionStatusChangeReason.Should().Be(ConnectionStatusChangeReason.Retry_Expired, $"Expected device to be {ConnectionStatusChangeReason.Retry_Expired} but was {lastConnectionStatusChangeReason}.");
+            connectionStatusChanges.Should().NotContainKey(ConnectionStatus.Disconnected_Retrying, $"Shouldn't get {ConnectionStatus.Disconnected_Retrying} status change.");
             int connected = connectionStatusChanges[ConnectionStatus.Connected];
-            Assert.AreEqual(1, connected, $"Should get {ConnectionStatus.Connected} once but was {connected}.");
+            connected.Should().Be(1, $"Should get {ConnectionStatus.Connected} once but got it {connected} times.");
             int disconnected = connectionStatusChanges[ConnectionStatus.Disconnected];
-            Assert.AreEqual(1, disconnected, $"Should get {ConnectionStatus.Disconnected} once but was {disconnected}.");
+            disconnected.Should().Be(1, $"Should get {ConnectionStatus.Disconnected} once but got it {disconnected} times.");
```

Comment:
I wanted to have a similar test for a custom retry policy, however, since we cannot simulate device-client-status-changing-transient-faults for a "duration", we cannot have an E2E test for this. It has been verified through a sample, though.